llvm/lib/Transforms/IPO/DynUntrustedAllocPost.cpp

```diff
@@ -338,17 +455,88 @@ class DynUntrustedAllocPost : public ModulePass {
     }
 
     auto allocHook = M.getFunction("allocHook");
+    //assert(allocHook != nullptr && "Cannot find allocHook for removal!");
     auto reallocHook = M.getFunction("reallocHook");
     auto deallocHook = M.getFunction("deallocHook");
 
+    // Working under the assumption that all missed cases of Hook calls is 
+    // due to the blocks containing them no longer being reachable, we remove
+    // those instructions from their respective blocks.
+    for (auto user : allocHook->users()) {
+      if (auto *inst = dyn_cast<Instruction>(user)) {
+        // TODO : Check to ensure instruction or block is unreachable
+        salvageDebugInfo(*inst);
+        inst->eraseFromParent();
+        total_hooks++;
+        hookCountMap.find("allocHook")->second++;
+        /* // Keeping this around in case necessary later for reference.
+        auto *func = inst->getFunction();
+        if(!func) {
+          errs() << "Following instruction did not have a parent function: " << *inst << "\n";
+          continue;
+        }
+        func->dump();
+        */
+      } else {
+        errs() << "User not an instruction: " << *user << "\n";
+        assert(false && "AllocHook user not an instruction!");
+      }
+    }
     allocHook->setLinkage(GlobalValue::LinkageTypes::InternalLinkage);
     allocHook->eraseFromParent();
 
+    for (auto user : reallocHook->users()) {
+      if (auto *inst = dyn_cast<Instruction>(user)) {
+       salvageDebugInfo(*inst);
+       inst->eraseFromParent();
+       total_hooks++;
+       hookCountMap.find("reallocHook")->second++;
+      } else {
+        errs() << "User not an instruction: " << *user << "\n";
+        assert(false && "ReAllocHook user not an instruction!");
+      }
+    }
     reallocHook->setLinkage(GlobalValue::LinkageTypes::InternalLinkage);
     reallocHook->eraseFromParent();
 
+    for (auto user : deallocHook->users()) {
+      if (auto *inst = dyn_cast<Instruction>(user)) {
+        salvageDebugInfo(*inst);
+        inst->eraseFromParent();
+        total_hooks++;
+        hookCountMap.find("deallocHook")->second++;
+      } else {
+        errs() << "User not an instruction: " << *user << "\n";
+        assert(false && "DeAllocHook user not an instruction!");
+      }
+    }
     deallocHook->setLinkage(GlobalValue::LinkageTypes::InternalLinkage);
     deallocHook->eraseFromParent();
+
+    // In addition to the hooks, conditionally check if mpk_SEGV_fault_handler
+    // is present and remove all uses and references.
+    auto mpkFaultHook = M.getFunction("mpk_SEGV_fault_handler");
```

Comment:
please extract some of these into utility functions.

Reply:
Removed code for `mpk_SEGV_fault_handler` removal as it did not function as intended (undefined references still prevent compilation from finishing).